### PR TITLE
feat(annotation-thread): added reply form, UI and other minor fixes

### PR DESCRIPTION
**Pull request by kkuliczkowski-box**
- Added reply form that is separated from the `ActivityThread`
- Improved UI to meet the design requirements
- Added `getUserProfileUrl` prop (and support for it) to `AnnotationThread`  in order to display linked user names

### CODE REVIEW

File: src/elements/content-sidebar/activity-feed/annotation-thread/useAnnotationThread.js [hunk 1]
```diff
@@ -98,6 +99,7 @@ const useAnnotationThread = ({
     };
 
     useAnnotatorEvents({
+        eventEmitter,
```
Comment (kkuliczkowski-box) [inline src/elements/content-sidebar/activity-feed/annotation-thread/useAnnotationThread.js:102]: I'm passing `eventEmitter` here in order for the current state of code to work, but I will create a seperate ticket for the refactoring of events handling as suggested by @jstoffan here: https://github.com/box/box-ui-elements/pull/3106#discussion_r1019439252

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThread.scss [hunk 2]
```diff
@@ -1,7 +1,6 @@
 @import '../../../../styles/variables';
 
 .AnnotationThread {
-    max-width: 340px;
-    max-height: 600px;
+    min-width: $bdl-grid-unit * 85;
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThread.scss:4]: We can probably revert this unless `85` is a meaningful value for our grid.

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.scss [hunk 3]
```diff
@@ -3,3 +3,22 @@
 .AnnotationThreadContent-loading {
     padding: $bdl-grid-unit * 5 0;
 }
+
+.AnnotationThread {
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.scss:7]: Should this be in `AnnotationThread.scss`?

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.scss [hunk 4]
```diff
@@ -3,3 +3,22 @@
...
+    .bcs-ActivityThread {
+        position: relative;
+        max-height: $bdl-grid-unit * 132;
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.scss:10]: Should we share this value with the other activity types? How are we keeping the threads and reply forms consistent?

File: src/elements/content-sidebar/activity-feed/annotation-thread/useAnnotationThread.js [hunk 5]
```diff
@@ -155,8 +157,10 @@ const useAnnotationThread = ({
     });
 
     React.useEffect(() => {
-        handleFetch({ annotationId, successCallback: handleFetchAnnotationSuccess });
-    }, [annotationId, handleFetch]);
+        if (!annotation || annotation.id !== annotationId) {
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/annotation-thread/useAnnotationThread.js:160]: Minor, but can we convert this to a guard clause?

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.js [hunk 6]
```diff
@@ -55,39 +59,49 @@ const AnnotationThreadContent = ({
...
+                    />
+                )}
+            </ActivityThread>
             {annotation && (
-                <AnnotationActivity
-                    getAvatarUrl={getAvatarUrl}
-                    currentUser={currentUser}
-                    isCurrentVersion
-                    item={annotation}
+                <ActivityThreadReplyForm
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.js:98]: Will we need to replicate this composition in each usage of `ActivityThread`? Should `ActivityThreadReplyForm` be used within that component, instead?

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.scss [hunk 7]
```diff
@@ -20,6 +20,10 @@
         padding: 0;
     }
 
+    .bcs-AnnotationActivity-menu {
+        top: 0;
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.scss:24]: What's the purpose behind this change?

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThread.scss (hunk 2 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThread.scss]: I don't think it's a meaningful value. I thought that the `$bdl-grid-unit` is a "base" grid size value and the width of elements should be connected to it. I've changed to a static value.

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.scss (hunk 3 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.scss]: Makes sense. I've moved it.

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.scss (hunk 4 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.scss]: This value sets the maximum height of the Activity Thread only when used in `AnnotationThread`. The reasoning behind it is that when Annotation is displayed in a "bubble" over the Canvas Board we don't want the thread to take too much vertical space. Similar limitation is not needed in other activity types that are displayed in Sidebar.
I've also changed the value to a static one.

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.js (hunk 6 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.js:98]: We could potentially refactor how the `ActivityThread` is used in Sidebar. Right now the `ActivityThreadReplyForm` is built as part of `ActivityThread` and is rendered if `onReplyCreate` handler is passed.

Here the `ActivityThreadReplyForm` needs to be displayed below the `ActivityThread` as the activity thread's height needs to be limited, but the form should always be visible.

File: src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.scss (hunk 7 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.scss:24]: When Annotation is rendered within the `ActivityThread` styling is a bit different and this change is to compensate that, so that the menu button is displayed in the same position.

As we haven't tested Annotations in Canvas before, we haven't caught that before.

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.js (hunk 6 above)
Comment (jstoffan) [reply src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.js:98]: I see. I think we could add a div around `children`, `renderButton()`, and the replies section [here](https://github.com/box/box-ui-elements/blob/master/src/elements/content-sidebar/activity-feed/activity-feed/ActivityThread.js#L113) that we can use to set the max-height and overflow values on. It's debatable, though, so not a blocking issue.

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThread.scss [hunk 8]
```diff
@@ -1,7 +1,23 @@
...
 
 .AnnotationThread {
-    max-width: 340px;
-    max-height: 600px;
+    min-width: 340px;
     background-color: $white;
+
+    .bcs-ActivityThread {
+        position: relative;
+        max-height: 528px;
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThread.scss:9]: Is this being derived from something? Is it `600px` minus some other value(s)? If so, we should have those values as variables, ideally.

**Review by jstoffan — APPROVED** [review]

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThread.scss (hunk 8 above)
Comment (kkuliczkowski-box) [reply src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThread.scss:13]: I don't think it's based on some other value.

As discussed, I've added `.bcs-ActivityThread-content` wrapper in the `ActivityThread`, so that the form is not used outside of it.

File: src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.js [hunk 9]
```diff
@@ -55,39 +58,43 @@ const AnnotationThreadContent = ({
...
-            isAlwaysExpanded
-            isRepliesLoading={isLoading}
-            mentionSelectorContacts={mentionSelectorContacts}
-            onReplyCreate={handleReplyCreate}
-            onReplyDelete={handleReplyDelete}
-            onReplyEdit={handleReplyEdit}
-            replies={replies}
-            repliesTotalCount={replies.length}
-        >
+        <>
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/annotation-thread/AnnotationThreadContent.js:61]: Should this be reverted?

**Review by jstoffan — APPROVED** [review]

**Review by wpiesiak — APPROVED** [review]
